src/registration.rs

```diff
@@ -100,46 +92,126 @@ impl Registrar {
             }
         };
 
-        let box_hostname = self.box_name.clone();
+        let client = Client::new();
+        let res = client.post(&self.registration_endpoint)
+            .header(Connection::close())
+            .body(&body)
+            .send();
+
+        // Sanity checks, mostly to debug errors since we don't try
+        // to recover from failures.
+        if let Ok(mut response) = res {
+            if response.status == StatusCode::Ok {
+                let mut body = String::new();
+                if let Ok(_) = response.read_to_string(&mut body) {
+                    info!("Server responded with: {}", body);
+                } else {
+                    info!("Unable to read answer from {}", self.registration_endpoint);
+                }
+            }
+        } else {
+            info!("Unable to send request to {}", self.registration_endpoint);
+        }
+    }
+
+    /// Register names with the DNS server (only registers the tunnel if the tunnel_frontend is non
+    /// None)
```

Comment:
Also explain that we create an A record to the box's local IP address and we also set two mirror A records to this address to avoid DNS propagation delays.